dspace-server-webapp/src/main/java/org/dspace/app/rest/hdlresolver/HdlResolverRestController.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.hdlresolver;
+
+import java.text.MessageFormat;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.handle.hdlresolver.HdlResolverDTO;
+import org.dspace.handle.hdlresolver.HdlResolverService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This controller is public and is useful for handle resolving,
+ * wheter a target handle identifier will be resolved into the
+ * corresponding URL (if found), otherwise will respond a null string.
+ * 
+ * @author Vincenzo Mecca (vins01-4science - vincenzo.mecca at 4science.it)
+ *
+ */
+@RestController
+@RequestMapping(path = "/{hdlService:hdlResolver|resolve|listhandles|listprefixes}/")
```

Comment:
As previously noted, this PR is missing a REST Contract or other Documentation to describe the behavior.  

Looking more closely at the code, it appears it creates 4 new endpoints outside of the `/api` path?
* `${dspace.server.url}/hdlResolver/{handle-prefix}/{handle-suffix}`
* `${dspace.server.url}/resolve/{handle-prefix}/{handle-suffix}`
* `${dspace.server.url}/listhandles/{handle-prefix}/{handle-suffix}`
* `${dspace.server.url}/listprefixes/{handle-prefix}/{handle-suffix}`

The examples of how to interact with these endpoints are lacking, as the tests are extremely basic.  They also all appear to do the same thing, which is to return the URL of whatever handle they are passed.

It's also unclear why the first endpoint is camel-cased (`hdlResolver`) instead of all lowercase (`hdlresolver`) as usual.

While I realize these are **not** traditional REST endpoints, the behavior here is odd to me & undocumented.  Could we consider adding these to our REST Contract?  Or if we feel they are better described elsewhere, maybe we could add details about these endpoints to the documentation for https://github.com/DSpace/Remote-Handle-Resolver ?

Reply:
> The code overall looks good to me. There are small things to be fixed in two comments.
> 
> @vins01-4science @abollini I have one general question: have you tested this with the RemoteHandleResolver the community offers? https://github.com/DSpace/Remote-Handle-Resolver This resolver sends request to three paths:
> 
> * BASE_URL/resolve/
> * BASE_URL/listhandles/
> * BASE_URL/listprefixes/
> 
> This PR does support the first URL only. Do you think you can add the other two? The code from DSpace 6 that supported this can be found here: https://github.com/DSpace/DSpace/blob/5e4aec74645ee0ef877952259347992d01353e84/dspace-jspui/src/main/java/org/dspace/app/webui/handle/HandleJSONResolver.java

Hello @tdonohue ,

the new endpoints have been suggested by @pnbecker.
Firstly the one and only endpoints was `hdlResolver`.

The behaviour is the same as in version 6, resolve an handle into a list of URLs.

Reply:
Hello @tdonohue , as said before, @pnbecker asked to extend the behaviour of the rest service that was responding under the path `hdlResolver` to work also with the other paths, then @abollini asked to add the paths, so here are the three paths.

For the lowercase I'll modify, but please let me know what to do with the other paths.

Thank you.

Reply:
@tdonohue : Thank you for the response, so as of now to let this PR be merged I should proceed with these changes?
- rename of `hdlResolver` to `hdlresolver`;
- remove the newly introduced endpoints: `resolve`, `listhandles`, `listprefixes`.

Is there something else?

Reply:
@vins01-4science : This PR cannot be merged until it is tested.  I'm unfortunately not able to test it as I don't have a remote handle server to use.

So, it's waiting on @pnbecker to test this with https://github.com/DSpace/Remote-Handle-Resolver.   Based on my reading of the code though, my suspicions are that we need to **keep** the `resolve`, `listhandles`, and `listprefixes` endpoints, *but they need to be fixed to act the same as they did in DSpace 6.x*.  Currently, as I've said their behavior looks to have changed.  However, I'm hoping that @pnbecker can verify whether my suspicions are correct (as it will require testing this PR with the `DSpace/Remote-Handle-Resolver` codebase.

If there is anyone at 4Science who has successfully tested this PR with the `DSpace/Remote-Handle-Resolver` code, they may also be able to answer my questions about the `resolve`, `listhandles` and `listprefixes` endpoints.

In the meantime, **yes** the `hdlResolver` should be renamed to `hdlresolver`.  That definitely needs to be fixed.

At this time, I'm uncertain how to move this PR forward until I can find a tester (so I'm hoping @pnbecker will find time to help).  I'm not comfortable with merging this without testing it, because of my questions about the `resolve`, `listhandles` and `listprefixes` endpoints.

Reply:
hi @tdonohue ,
there was an internal misunderstand.
You are right the three endpoints have different purposes and it is required to restore their proper behavior.
More specifically, 
- the `/resolve' endpoint is needed to allow the resolution of an exact handle (the behavior implemented so far)
- the `/listprefixes` endpoint is needed to allow the handle server to discover which prefixes are managed by a specific dspace instance, as the ultimate goal of the  `DSpace/Remote-Handle-Resolve` is to allow a single handle server to serve multiple dspace instances
- the `listhandles` endpoint was required by the internal handle system interface but I'm not sure if it is really used in normal operation.

In short, this PR needs to be updated to fix the behavior of the three different endpoints.
It should be possible to test this PR also without a real handle server as you can start the handle server locally configuring it accordingly to the documentation in the `DSpace/Remote-Handle-Resolver` and access it local http interface at port 8000. From here you will be able to query your fake handle and see that the handle server will talk with the /resolve endpoint of DSpace. The /listprefixes endpoint of DSpace is instead call when the handle server start